incubator/vault/README.md

```diff
@@ -0,0 +1,69 @@
+# Vault Helm Chart
+
+This directory contains a Kubernetes chart to deploy a Vault server.
+
+## Prerequisites Details
+
+* Kubernetes 1.5
+
+## Chart Details
+
+This chart will do the following:
+
+* Implement a Vault deployment
+
+Please note that a backend service for Vault (for example, Consul) must
+be deployed beforehand and configured with the `vault` option. YAML provided
+under this option will be converted to JSON for the final vault `config.json`
+file.
+
+> See https://www.vaultproject.io/docs/configuration/ for more information.
+
+## Installing the Chart
```

Comment:
Hashicorp recommends running Vault on a [dedicated instance](https://www.vaultproject.io/guides/production.html) to reduce the risk of the machine running Vault getting rooted, and thus allowing an attacker to get to unencrypted secrets. Can you add links to documentation like [kops's instance groups](https://github.com/kubernetes/kops/blob/master/docs/instance_groups.md) and an example config on how to isolate pods to those nodes?

Reply:
I think running an entire VM instance just for the one container that is running (a rootfs read-only) vault is a little overkill at that point. :-)

Reply:
The risk here is that if Vault is running on the same machine as some other container that's able to get root on the system, it can read the Vault secrets directly out of memory. I've seen some Vault setups that go even more extreme than machine-isolation. For example, Nike created [Cerberus](http://engineering.nike.com/cerberus) that wraps around Vault, and they recommend isolating their setup off in a [dedicated AWS account](http://engineering.nike.com/cerberus/docs/administration-guide/creating-an-environment). I don't think we should hold up this chart for this documentation, but I think it'd be worthwhile adding it in the future.